### DISCUSSION TITLE
Run remotely, see results locally?

### DISCUSSION THREAD

**Discussion by cifkao** [kiyoon/jupynium.nvim#112] [Q&A]
Jupynium looks like a really cool project and it seemed like it might be able to give me the same experience as running Python code cells in a remote VSCode session, where the code runs in a remote environment, but the results show up locally.

However, I seem to have misunderstood how this works. Steps I followed:
- In my remote environment:
  - `pip install jupynium`
  - `nvim --listen 0.0.0.0:18898 test.ju.py`
- Local:
  - `pip install jupynium`
  - `jupynium --nvim_listen_addr remote:18898`
- Remote:
  - `:JupyniumStartSync`
  - ```python
    # %%
    !hostname
    ```
- Now I expected the code to run on the remote machine, but it instead runs on my local machine.

Is there a way to get the experience that I'm looking for?

**Comment by kiyoon** [comment]
For this to work, you should open the Jupyter Notebook manually on the remote machine, and port forward it so that `localhost:8888` connects to the remote Jupyter Notebook.

**Reply by cifkao** [reply]
Right, but you're just talking about plain Jupyter notebooks? Or is it still possible to control Jupyter from neovim in this case?

**Reply by kiyoon** [reply]
Jupynium controls Jupyter Notebook through the browser. In other words it doesn't need an external extension. Once it loads Jupyter Notebook from the browser it can control it.

As a convenience feature, it also start Jupyter Notebook automatically (but locally only) if it can't connect to the `localhost:8888`.
